web-server/v0.4/src/routes/Dashboard/RunComparison.js

```diff
@@ -303,6 +331,126 @@ export default class RunComparison extends ReactJS.Component {
     this.setState({ timeseriesDropdownSelected: timeseriesDropdownSelected });
   };
 
+  savePDF = async () => {
+    /* maps through the state array containing the IDs and saves the pdf */
+    var doc = new jsPDF('p', 'mm', [1050, 1475]);
+    var position = 5;
+    var fullIHeight = 0;
+    var promises = this.state.defaultChecked.map(async (check, index) => {
+      await html2canvas(document.querySelector('#' + check), { allowTaint: true }).then(canvas => {
+        doc.setPage(index);
+        canvas.getContext('2d');
+        var imgData = canvas.toDataURL();
+        var imgWidth = 345;
+        var pageHeight = 510;
+        var imgHeight = (canvas.height * imgWidth) / canvas.width;
+        position = fullIHeight + 5;
+        fullIHeight = fullIHeight + imgHeight;
+        var heightLeft = fullIHeight;
+        doc = doc.addImage(imgData, 'PNG', 10, position, imgWidth, imgHeight);
+        doc.text(this.state.pdfHeader, 172, 10, 'center');
+        heightLeft -= pageHeight;
+        while (heightLeft >= 0) {
```

Comment:
You introduced a while statement here to offset the position of the document while adding an image; however, it might make sense to remove this while statement as you assign the `heightLeft` just before the loop.